src/index.js

```diff
@@ -1,44 +1,104 @@
 
-const section = document.querySelector('section');
+/* All HTML References */
 const taskInput = document.querySelector('input');
 const tasksContainer = document.querySelector('.tasks-container');
-const allTasksContainer = document.getElementById('tasks-container');
 const completedTasks = document.querySelector('button');
 
-/* class myList{
-    constructor(description, completed, index){
-        this.description = description;
-        this.completed = completed;
-        this.index = index;
-    }
+/* Class Object */
+class MyList {
+  constructor(description, completed, index) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
   }
-  */
-const list = [
-  {
-    description: 'wash dishes',
-    completed: true,
-    index: 1,
-  },
-  {
-    description: 'make the bed',
-    completed: true,
-    index: 2,
-  },
-];
-  /* const listArray = []; */
-list.forEach((Element) => {
+}
+
+const listArray = [];
+export default listArray;
+const addTask = (taskValue) => {
   const newTaskContainer = document.createElement('div');
   newTaskContainer.classList.add('new-task--container');
   newTaskContainer.innerHTML += `
-   <div class="left-content">
-      <input type="checkbox" ${Element.completed} class="check-task">
-      <span>${Element.description}</span>
-   </div>
-   <div class="right-content">
-      <i class="fa-solid fa-ellipsis-vertical"></i>
-      <i class="fa-solid fa-trash-can"></i>
-   </div>`;
-  allTasksContainer.appendChild(newTaskContainer);
-});
\ No newline at end of file
+    <input type="checkbox" class="check-task">
+    <span>${taskValue}</span>
+    <i class="fa-solid fa-ellipsis-vertical"></i>
+    <i class="fa-solid fa-trash-can"></i>
+  `;
+  tasksContainer.appendChild(newTaskContainer);
+
+  const checkbox = document.querySelectorAll('.check-task');
+  /* VAMOS AQUÍ */
+  checkbox.forEach((i) => {
+    i.addEventListener('click', () => {
+      i.parentElement.classList.toggle('checked-container');
+      i.nextElementSibling.classList.toggle('task-checked');
+      updatingData();
+    });
+  });
+
+  const storageObject = new MyList(taskValue, false, checkbox.length - 1);
+  listArray.push(storageObject);
+  localStorage.setItem('list', JSON.stringify(listArray));
+};
+/* Adding task Event */
+taskInput.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter' && taskInput.value) {
+    addTask(taskInput.value);
+    taskInput.value = null;
+  }
+});
+
+const takeData = () => {
+  const data = JSON.parse(localStorage.getItem('list'));
+  data.map((i) => {
+    listArray.push(i);
+    const newTaskContainer = document.createElement('div');
+    newTaskContainer.classList.add('new-task--container');
+    newTaskContainer.innerHTML += `
+    <input type="checkbox" class="check-task">
+    <span>${i.description}</span>
+    <i class="fa-solid fa-ellipsis-vertical"></i>
+    <i class="fa-solid fa-trash-can"></i>
+    `;
```

Comment:
- [ ] Kindly you have to check here if the task is completed to make the check box checked and add a line through the task description because if you marked a task as completed and reload the check box of this task will be unchecked and the task description won't have a line through it.